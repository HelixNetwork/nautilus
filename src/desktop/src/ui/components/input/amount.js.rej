diff a/src/desktop/src/ui/components/input/amount.js b/src/desktop/src/ui/components/input/amount.js	(rejected hunks)
@@ -1,6 +1,6 @@
-import React from "react";
-import PropTypes from "prop-types";
-import classNames from "classnames";
-import { formatHlx, TOTAL_HELIX_SUPPLY } from "libs/hlx/utils";
-import { round } from "libs/utils";
-import { getCurrencySymbol, formatMonetaryValue } from "libs/currency";
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { formatHlx, TOTAL_HELIX_SUPPLY } from 'libs/hlx/utils';
+import { round } from 'libs/utils';
+import { getCurrencySymbol, formatMonetaryValue } from 'libs/currency';
@@ -8,2 +8,2 @@ import { getCurrencySymbol, formatMonetaryValue } from "libs/currency";
-import Icon from "ui/components/icon";
-import css from "./input.scss";
+import Icon from 'ui/components/icon';
+import css from './input.scss';
@@ -11 +11 @@ import css from "./input.scss";
-const units = ["HLX", "kHLX", "mHLX", "gHLX", "tHLX", "$"];
+const units = ['HLX', 'kHLX', 'mHLX', 'gHLX', 'tHLX', '$'];
@@ -18,50 +18,35 @@ export default class AmountInput extends React.PureComponent {
-  static propTypes = {
-    /** Current amount value */
-    amount: PropTypes.string.isRequired,
-    /** Element id */
-    id: PropTypes.string,
-    /** Max available amount */
-    balance: PropTypes.number.isRequired,
-    /** Fiat currency settings
-     * @property {string} conversionRate - Active currency conversion rate to hlx
-     * @property {string} currency - Active currency name
-     * @property {string} usdPrice - Current USD price per hlx
-     */
-    settings: PropTypes.shape({
-      conversionRate: PropTypes.number.isRequired,
-      currency: PropTypes.string.isRequired,
-      usdPrice: PropTypes.number.isRequired
-    }).isRequired,
-    /** Amount input label */
-    label: PropTypes.string.isRequired,
-    /** Max amount controller label */
-    labelMax: PropTypes.string.isRequired,
-    /** Amount change event function
-     * @param {string} value - Current amount value
-     */
-    onChange: PropTypes.func.isRequired
-  };
-
-  state = {
-    unit: "mHLX",
-    value: 0,
-    helix: 0
-  };
-
-  componentWillMount() {
-    this.stateToProps(this.props);
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.stateToProps(nextProps);
-  }
-
-  onChange = value => {
-    // Replace , with . and remove leading zeros
-    value = value.replace(/,/g, ".").replace(/^0+(?=\d)/, "");
-
-    // Ignore invalid or empty input
-    if (!value.length) {
-      value = "0";
-    } else if (!value.match(/^\d+(\.\d{0,20})?$/g)) {
-      return;
+    static propTypes = {
+        /** Current amount value */
+        amount: PropTypes.string.isRequired,
+        /** Element id */
+        id: PropTypes.string,
+        /** Max available amount */
+        balance: PropTypes.number.isRequired,
+        /** Fiat currency settings
+         * @property {string} conversionRate - Active currency conversion rate to hlx
+         * @property {string} currency - Active currency name
+         * @property {string} usdPrice - Current USD price per hlx
+         */
+        settings: PropTypes.shape({
+            conversionRate: PropTypes.number.isRequired,
+            currency: PropTypes.string.isRequired,
+            usdPrice: PropTypes.number.isRequired,
+        }).isRequired,
+        /** Amount input label */
+        label: PropTypes.string.isRequired,
+        /** Max amount controller label */
+        labelMax: PropTypes.string.isRequired,
+        /** Amount change event function
+         * @param {string} value - Current amount value
+         */
+        onChange: PropTypes.func.isRequired,
+    };
+
+    state = {
+        unit: 'mHLX',
+        value: 0,
+        helix: 0,
+    };
+
+    componentWillMount() {
+        this.stateToProps(this.props);
@@ -70,6 +55,2 @@ export default class AmountInput extends React.PureComponent {
-    // Strip to max allowed decimals
-    if (value.indexOf(".") > 0) {
-      value = value.substr(
-        0,
-        value.indexOf(".") + decimals[units.indexOf(this.state.unit)] + 1
-      );
+    componentWillReceiveProps(nextProps) {
+        this.stateToProps(nextProps);
@@ -78,51 +59,55 @@ export default class AmountInput extends React.PureComponent {
-    // Convert to helix
-    const helix = round(value * this.getUnitMultiplier());
-
-    if (helix > TOTAL_HELIX_SUPPLY) {
-      return;
-    }
-
-    this.setState(
-      {
-        value: value,
-        helix: helix
-      },
-      () => this.props.onChange(String(helix))
-    );
-  };
-
-  getUnitMultiplier(unit) {
-    let multiplier = 1;
-    const target = unit || this.state.unit;
-    switch (target) {
-      default :
-        break;
-      case "kHLX":
-        multiplier = 1000;
-        break;
-      case "mHLX":
-        multiplier = 1000000;
-        break;
-      case "gHLX":
-        multiplier = 1000000000;
-        break;
-      case "tHLX":
-        multiplier = 1000000000000;
-        break;
-      case "$":
-        multiplier =
-          1000000 /
-          (this.props.settings.usdPrice * this.props.settings.conversionRate);
-        break;
-    }
-    return multiplier;
-  }
-
-  stateToProps = props => {
-    if (this.state.helix !== parseInt(props.amount)) {
-      this.setState({
-        helix: props.amount.length ? parseInt(props.amount) : 0,
-        value: props.amount.length
-          ? parseInt(props.amount) / this.getUnitMultiplier()
-          : 0
-      });
+    onChange = (value) => {
+        // Replace , with . and remove leading zeros
+        value = value.replace(/,/g, '.').replace(/^0+(?=\d)/, '');
+
+        // Ignore invalid or empty input
+        if (!value.length) {
+            value = '0';
+        } else if (!value.match(/^\d+(\.\d{0,20})?$/g)) {
+            return;
+        }
+
+        // Strip to max allowed decimals
+        if (value.indexOf('.') > 0) {
+            value = value.substr(0, value.indexOf('.') + decimals[units.indexOf(this.state.unit)] + 1);
+        }
+
+        // Convert to helix
+        const helix = round(value * this.getUnitMultiplier());
+
+        if (helix > TOTAL_HELIX_SUPPLY) {
+            return;
+        }
+
+        this.setState(
+            {
+                value: value,
+                helix: helix,
+            },
+            () => this.props.onChange(String(helix)),
+        );
+    };
+
+    getUnitMultiplier(unit) {
+        let multiplier = 1;
+        const target = unit || this.state.unit;
+        switch (target) {
+            default:
+                break;
+            case 'kHLX':
+                multiplier = 1000;
+                break;
+            case 'mHLX':
+                multiplier = 1000000;
+                break;
+            case 'gHLX':
+                multiplier = 1000000000;
+                break;
+            case 'tHLX':
+                multiplier = 1000000000000;
+                break;
+            case '$':
+                multiplier = 1000000 / (this.props.settings.usdPrice * this.props.settings.conversionRate);
+                break;
+        }
+        return multiplier;
@@ -130 +114,0 @@ export default class AmountInput extends React.PureComponent {
-  };
@@ -132,47 +116,82 @@ export default class AmountInput extends React.PureComponent {
-  maxAmount = () => {
-    const { amount, balance } = this.props;
-    const total = balance === parseInt(amount) ? 0 : balance;
-    this.props.onChange(String(total));
-  };
-
-  unitChange = unit => {
-    if (unit === this.state.unit) {
-      return;
-    }
-
-    this.setState(prevState => {
-      const helix = round(prevState.value * this.getUnitMultiplier(unit));
-
-      const value =
-        helix <= TOTAL_HELIX_SUPPLY
-          ? prevState.value
-          : round(TOTAL_HELIX_SUPPLY / this.getUnitMultiplier(unit));
-
-      this.props.onChange(String(helix));
-
-      return {
-        unit: unit,
-        helix: helix,
-        value: value
-      };
-    });
-  };
-
-  render() {
-    const { amount, balance, id, settings, label, labelMax } = this.props;
-    const { value, unit } = this.state;
-
-    return (
-      <div className={css.input}>
-        <fieldset>
-          <a>
-            <strong>
-              {unit === "$" ? getCurrencySymbol(settings.currency) : unit}
-              <Icon icon="chevronDown" size={8} />
-              <ul className={css.dropdown}>
-                {units.map(item => {
-                  return (
-                    <li
-                      key={item}
-                      className={item === unit ? css.selected : null}
-                      onClick={() => this.unitChange(item)}
+    stateToProps = (props) => {
+        if (this.state.helix !== parseInt(props.amount)) {
+            this.setState({
+                helix: props.amount.length ? parseInt(props.amount) : 0,
+                value: props.amount.length ? parseInt(props.amount) / this.getUnitMultiplier() : 0,
+            });
+        }
+    };
+
+    maxAmount = () => {
+        const { amount, balance } = this.props;
+        const total = balance === parseInt(amount) ? 0 : balance;
+        this.props.onChange(String(total));
+    };
+
+    unitChange = (unit) => {
+        if (unit === this.state.unit) {
+            return;
+        }
+
+        this.setState((prevState) => {
+            const helix = round(prevState.value * this.getUnitMultiplier(unit));
+
+            const value =
+                helix <= TOTAL_HELIX_SUPPLY
+                    ? prevState.value
+                    : round(TOTAL_HELIX_SUPPLY / this.getUnitMultiplier(unit));
+
+            this.props.onChange(String(helix));
+
+            return {
+                unit: unit,
+                helix: helix,
+                value: value,
+            };
+        });
+    };
+
+    render() {
+        const { amount, balance, id, settings, label, labelMax } = this.props;
+        const { value, unit } = this.state;
+
+        return (
+            <div className={css.input}>
+                <fieldset>
+                    <a>
+                        <strong>
+                            {unit === '$' ? getCurrencySymbol(settings.currency) : unit}
+                            <Icon icon="chevronDown" size={8} />
+                            <ul className={css.dropdown}>
+                                {units.map((item) => {
+                                    return (
+                                        <li
+                                            key={item}
+                                            className={item === unit ? css.selected : null}
+                                            onClick={() => this.unitChange(item)}
+                                        >
+                                            {item === '$' ? getCurrencySymbol(settings.currency) : item}
+                                        </li>
+                                    );
+                                })}
+                            </ul>
+                        </strong>
+                        {amount > 0 && unit !== '$' ? (
+                            <p>
+                                ={' '}
+                                {formatMonetaryValue(
+                                    amount,
+                                    settings.usdPrice * settings.conversionRate,
+                                    settings.currency,
+                                )}
+                            </p>
+                        ) : null}
+                        {amount > 0 && unit === '$' ? <p>= {formatHlx(amount, false, true)}</p> : null}
+                    </a>
+                    <input id={id} type="text" value={value} onChange={(e) => this.onChange(e.target.value)} />
+                    <small>{label}</small>
+                </fieldset>
+                {balance > 0 && (
+                    <a
+                        className={classNames(css.checkbox, parseInt(amount) === balance ? css.on : css.off)}
+                        onClick={this.maxAmount}
@@ -180,15 +199,2 @@ export default class AmountInput extends React.PureComponent {
-                      {item === "$"
-                        ? getCurrencySymbol(settings.currency)
-                        : item}
-                    </li>
-                  );
-                })}
-              </ul>
-            </strong>
-            {amount > 0 && unit !== "$" ? (
-              <p>
-                ={" "}
-                {formatMonetaryValue(
-                  amount,
-                  settings.usdPrice * settings.conversionRate,
-                  settings.currency
+                        {labelMax}
+                    </a>
@@ -196,28 +202,3 @@ export default class AmountInput extends React.PureComponent {
-              </p>
-            ) : null}
-            {amount > 0 && unit === "$" ? (
-              <p>= {formatHlx(amount, false, true)}</p>
-            ) : null}
-          </a>
-          <input
-            id={id}
-            type="text"
-            value={value}
-            onChange={e => this.onChange(e.target.value)}
-          />
-          <small>{label}</small>
-        </fieldset>
-        {balance > 0 && (
-          <a
-            className={classNames(
-              css.checkbox,
-              parseInt(amount) === balance ? css.on : css.off
-            )}
-            onClick={this.maxAmount}
-          >
-            {labelMax}
-          </a>
-        )}
-      </div>
-    );
-  }
+            </div>
+        );
+    }
