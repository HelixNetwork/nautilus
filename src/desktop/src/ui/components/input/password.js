import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withI18n } from 'react-i18next';
import { zxcvbn } from 'libs/exports';

import { passwordReasons } from 'libs/password';

import Icon from 'ui/components/icon';
import css from './input.scss';

/**
 * Password input component
 */
export class PasswordComponent extends React.PureComponent {

    render() {
        return (
            <div>Hi</div>
        )
    }
}
export default ((withI18n)(PasswordComponent));