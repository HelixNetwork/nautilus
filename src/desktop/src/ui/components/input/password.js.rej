diff a/src/desktop/src/ui/components/input/password.js b/src/desktop/src/ui/components/input/password.js	(rejected hunks)
@@ -1,5 +1,5 @@
-import React from "react";
-import PropTypes from "prop-types";
-import classNames from "classnames";
-import { withI18n } from "react-i18next";
-import { zxcvbn } from "libs/exports";
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { withI18n } from 'react-i18next';
+import { zxcvbn } from 'libs/exports';
@@ -7 +7 @@ import { zxcvbn } from "libs/exports";
-import { passwordReasons } from "libs/password";
+import { passwordReasons } from 'libs/password';
@@ -9,2 +9,2 @@ import { passwordReasons } from "libs/password";
-import Icon from "ui/components/icon";
-import css from "./input.scss";
+import Icon from 'ui/components/icon';
+import css from './input.scss';
@@ -16,25 +16,25 @@ class PasswordInput extends React.PureComponent {
-  static propTypes = {
-    /** Current password value */
-    value: PropTypes.string.isRequired,
-    /** Should input focus when changed to true */
-    focus: PropTypes.bool,
-    /** Is the component disabled */
-    disabled: PropTypes.bool,
-    /** Should input show the password strength score */
-    showScore: PropTypes.bool,
-    /** Should input show the validation checkmark */
-    showValid: PropTypes.bool,
-    /** Should the input match a string */
-    match: PropTypes.string,
-    /** Password input label */
-    label: PropTypes.string.isRequired,
-    /** Password change event function
-     * @param {string} value - Current password value
-     */
-    onChange: PropTypes.func.isRequired,
-    /** Translation helper
-     * @param {string} translationString - locale string identifier to be translated
-     * @ignore
-     */
-    t: PropTypes.func.isRequired
-  };
+    static propTypes = {
+        /** Current password value */
+        value: PropTypes.string.isRequired,
+        /** Should input focus when changed to true */
+        focus: PropTypes.bool,
+        /** Is the component disabled */
+        disabled: PropTypes.bool,
+        /** Should input show the password strength score */
+        showScore: PropTypes.bool,
+        /** Should input show the validation checkmark */
+        showValid: PropTypes.bool,
+        /** Should the input match a string */
+        match: PropTypes.string,
+        /** Password input label */
+        label: PropTypes.string.isRequired,
+        /** Password change event function
+         * @param {string} value - Current password value
+         */
+        onChange: PropTypes.func.isRequired,
+        /** Translation helper
+         * @param {string} translationString - locale string identifier to be translated
+         * @ignore
+         */
+        t: PropTypes.func.isRequired,
+    };
@@ -42,4 +42,4 @@ class PasswordInput extends React.PureComponent {
-  state = {
-    hidden: true,
-    capsLock: false
-  };
+    state = {
+        hidden: true,
+        capsLock: false,
+    };
@@ -47,3 +47,7 @@ class PasswordInput extends React.PureComponent {
-  componentDidMount() {
-    if (this.props.focus) {
-      this.input.focus();
+    componentDidMount() {
+        if (this.props.focus) {
+            this.input.focus();
+        }
+        this.onKey = this.setCapsLock.bind(this);
+        window.addEventListener('keydown', this.onKey);
+        window.addEventListener('keyup', this.onKey);
@@ -51,4 +54,0 @@ class PasswordInput extends React.PureComponent {
-    this.onKey = this.setCapsLock.bind(this);
-    window.addEventListener("keydown", this.onKey);
-    window.addEventListener("keyup", this.onKey);
-  }
@@ -56,3 +56,4 @@ class PasswordInput extends React.PureComponent {
-  componentWillReceiveProps(nextProps) {
-    if (!this.props.focus && nextProps.focus) {
-      this.input.focus();
+    componentWillReceiveProps(nextProps) {
+        if (!this.props.focus && nextProps.focus) {
+            this.input.focus();
+        }
@@ -60 +60,0 @@ class PasswordInput extends React.PureComponent {
-  }
@@ -62,4 +62,4 @@ class PasswordInput extends React.PureComponent {
-  componentWillUnmount() {
-    window.removeEventListener("keydown", this.onKey);
-    window.removeEventListener("keyup", this.onKey);
-  }
+    componentWillUnmount() {
+        window.removeEventListener('keydown', this.onKey);
+        window.removeEventListener('keyup', this.onKey);
+    }
@@ -67,5 +67,5 @@ class PasswordInput extends React.PureComponent {
-  setVisibility = () => {
-    this.setState(prevState => ({
-      hidden: !prevState.hidden
-    }));
-  };
+    setVisibility = () => {
+        this.setState((prevState) => ({
+            hidden: !prevState.hidden,
+        }));
+    };
@@ -73,5 +73,5 @@ class PasswordInput extends React.PureComponent {
-  setCapsLock = e => {
-    this.setState({
-      capsLock: e.getModifierState("CapsLock")
-    });
-  };
+    setCapsLock = (e) => {
+        this.setState({
+            capsLock: e.getModifierState('CapsLock'),
+        });
+    };
@@ -79,12 +79,3 @@ class PasswordInput extends React.PureComponent {
-  render() {
-    const {
-      label,
-      value,
-      disabled,
-      onChange,
-      showScore,
-      showValid,
-      match,
-      t
-    } = this.props;
-    const { hidden, capsLock } = this.state;
+    render() {
+        const { label, value, disabled, onChange, showScore, showValid, match, t } = this.props;
+        const { hidden, capsLock } = this.state;
@@ -92,52 +83,3 @@ class PasswordInput extends React.PureComponent {
-    const score = zxcvbn(value);
-    const isValid =
-      score.score === 4 && (typeof match !== "string" || match === value);
-    return (
-      <div
-        className={classNames(
-          css.input,
-          css.padded,
-          disabled ? css.disabled : null,
-          capsLock ? css.capsLock : null
-        )}
-        style={this.props.style ? this.props.style : {}}
-      >
-        <fieldset>
-          <a
-            className={hidden ? css.strike : null}
-            onClick={this.setVisibility}
-          >
-            <Icon icon="eye" size={16} />
-          </a>
-          <input
-            type={hidden ? "password" : "text"}
-            ref={input => {
-              this.input = input;
-            }}
-            value={value}
-            onClick={this.setCapsLock}
-            onKeyDown={this.setCapsLock}
-            onChange={e => onChange(e.target.value)}
-          />
-          <small>{label}</small>
-          <strong>
-            {/* <Icon icon="attention" size={14} /> */}
-            {t("capsLockIsOn")}
-          </strong>
-          {showScore ? (
-            <React.Fragment>
-              <div className={css.score} data-strength={score.score}>
-                <span />
-                <span />
-                <span />
-              </div>
-              {value.length > 8 && passwordReasons[score.feedback.warning] ? (
-                <div className={css.hint}>
-                  {t(
-                    `changePassword:${passwordReasons[score.feedback.warning]}`
-                  )}
-                </div>
-              ) : null}
-            </React.Fragment>
-          ) : null}
-          {showValid ? (
+        const score = zxcvbn(value);
+        const isValid = score.score === 4 && (typeof match !== 'string' || match === value);
+        return (
@@ -145 +87,7 @@ class PasswordInput extends React.PureComponent {
-              className={classNames(css.valid, isValid ? css.isValid : null)}
+                className={classNames(
+                    css.input,
+                    css.padded,
+                    disabled ? css.disabled : null,
+                    capsLock ? css.capsLock : null,
+                )}
+                style={this.props.style ? this.props.style : {}}
@@ -147 +95,39 @@ class PasswordInput extends React.PureComponent {
-              <Icon icon="tickRound" size={26} />
+                <fieldset>
+                    <a className={hidden ? css.strike : null} onClick={this.setVisibility}>
+                        <Icon icon="eye" size={16} />
+                    </a>
+                    <input
+                        type={hidden ? 'password' : 'text'}
+                        ref={(input) => {
+                            this.input = input;
+                        }}
+                        value={value}
+                        onClick={this.setCapsLock}
+                        onKeyDown={this.setCapsLock}
+                        onChange={(e) => onChange(e.target.value)}
+                    />
+                    <small>{label}</small>
+                    <strong>
+                        {/* <Icon icon="attention" size={14} /> */}
+                        {t('capsLockIsOn')}
+                    </strong>
+                    {showScore ? (
+                        <React.Fragment>
+                            <div className={css.score} data-strength={score.score}>
+                                <span />
+                                <span />
+                                <span />
+                            </div>
+                            {value.length > 8 && passwordReasons[score.feedback.warning] ? (
+                                <div className={css.hint}>
+                                    {t(`changePassword:${passwordReasons[score.feedback.warning]}`)}
+                                </div>
+                            ) : null}
+                        </React.Fragment>
+                    ) : null}
+                    {showValid ? (
+                        <div className={classNames(css.valid, isValid ? css.isValid : null)}>
+                            <Icon icon="tickRound" size={26} />
+                        </div>
+                    ) : null}
+                </fieldset>
@@ -149,5 +135,2 @@ class PasswordInput extends React.PureComponent {
-          ) : null}
-        </fieldset>
-      </div>
-    );
-  }
+        );
+    }
