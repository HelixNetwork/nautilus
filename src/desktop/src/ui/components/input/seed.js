import React from 'react';
import PropTypes from 'prop-types';

/*
 * Seed input component
 */
export class SeedComponent extends React.PureComponent {
    static propTypes = {
        seed: PropTypes.array.isRequired,
        label: PropTypes.string.isRequired,
        focus: PropTypes.bool,
        closeLabel: PropTypes.string.isRequired,
        onChange: PropTypes.func.isRequired,
        setAccountInfoDuringSetup: PropTypes.func.isRequired,
        updateImportName: PropTypes.bool,
        generateAlert: PropTypes.func.isRequired,
        t: PropTypes.func.isRequired,
    };

    state = {
        showScanner: false,
        importBuffer: null,
        hidden: true,
        cursor: 0,
        accounts: [],
        accountIndex: -1,
    };

    componentDidMount() {
        if (this.props.focus) {
            this.input.focus();
        }
    }

    componentWillReceiveProps(nextProps) {
        if (!this.props.focus && nextProps.focus) {
            this.input.focus();
        }
    }

    componentDidUpdate() {
        if (this.input && this.props.seed && this.props.seed.length >= this.state.cursor) {
            try {
                const range = document.createRange();
                const sel = window.getSelection();
                range.setStart(this.input, this.state.cursor);
                range.collapse(true);
                sel.removeAllRanges();
                sel.addRange(range);
                this.input.scrollLeft = range.startOffset * 10;
            } catch (error) {}
        }
    }
}
