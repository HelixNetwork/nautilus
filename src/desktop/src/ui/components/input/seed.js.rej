diff a/src/desktop/src/ui/components/input/seed.js b/src/desktop/src/ui/components/input/seed.js	(rejected hunks)
@@ -1,2 +1,2 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React from 'react';
+import PropTypes from 'prop-types';
@@ -8,11 +8,11 @@ export class SeedComponent extends React.PureComponent {
-  static propTypes = {
-    seed: PropTypes.array.isRequired,
-    label: PropTypes.string.isRequired,
-    focus: PropTypes.bool,
-    closeLabel: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
-    setAccountInfoDuringSetup: PropTypes.func.isRequired,
-    updateImportName: PropTypes.bool,
-    generateAlert: PropTypes.func.isRequired,
-    t: PropTypes.func.isRequired
-  };
+    static propTypes = {
+        seed: PropTypes.array.isRequired,
+        label: PropTypes.string.isRequired,
+        focus: PropTypes.bool,
+        closeLabel: PropTypes.string.isRequired,
+        onChange: PropTypes.func.isRequired,
+        setAccountInfoDuringSetup: PropTypes.func.isRequired,
+        updateImportName: PropTypes.bool,
+        generateAlert: PropTypes.func.isRequired,
+        t: PropTypes.func.isRequired,
+    };
@@ -20,8 +20,8 @@ export class SeedComponent extends React.PureComponent {
-  state = {
-    showScanner: false,
-    importBuffer: null,
-    hidden: true,
-    cursor: 0,
-    accounts: [],
-    accountIndex: -1
-  };
+    state = {
+        showScanner: false,
+        importBuffer: null,
+        hidden: true,
+        cursor: 0,
+        accounts: [],
+        accountIndex: -1,
+    };
@@ -29,3 +29,4 @@ export class SeedComponent extends React.PureComponent {
-  componentDidMount() {
-    if (this.props.focus) {
-      this.input.focus();
+    componentDidMount() {
+        if (this.props.focus) {
+            this.input.focus();
+        }
@@ -33 +33,0 @@ export class SeedComponent extends React.PureComponent {
-  }
@@ -35,3 +35,4 @@ export class SeedComponent extends React.PureComponent {
-  componentWillReceiveProps(nextProps) {
-    if (!this.props.focus && nextProps.focus) {
-      this.input.focus();
+    componentWillReceiveProps(nextProps) {
+        if (!this.props.focus && nextProps.focus) {
+            this.input.focus();
+        }
@@ -39 +39,0 @@ export class SeedComponent extends React.PureComponent {
-  }
@@ -41,15 +41,12 @@ export class SeedComponent extends React.PureComponent {
-  componentDidUpdate() {
-    if (
-      this.input &&
-      this.props.seed &&
-      this.props.seed.length >= this.state.cursor
-    ) {
-      try {
-        const range = document.createRange();
-        const sel = window.getSelection();
-        range.setStart(this.input, this.state.cursor);
-        range.collapse(true);
-        sel.removeAllRanges();
-        sel.addRange(range);
-        this.input.scrollLeft = range.startOffset * 10;
-      } catch (error) {}
+    componentDidUpdate() {
+        if (this.input && this.props.seed && this.props.seed.length >= this.state.cursor) {
+            try {
+                const range = document.createRange();
+                const sel = window.getSelection();
+                range.setStart(this.input, this.state.cursor);
+                range.collapse(true);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                this.input.scrollLeft = range.startOffset * 10;
+            } catch (error) {}
+        }
@@ -57 +53,0 @@ export class SeedComponent extends React.PureComponent {
-  }
