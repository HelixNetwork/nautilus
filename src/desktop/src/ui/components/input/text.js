import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import css from './input.scss';

/**
 * Single line input component
 */
export default class Text extends React.PureComponent {
    static propTypes = {
        value: PropTypes.string.isRequired,
        disabled: PropTypes.bool,
        focus: PropTypes.bool,
        label: PropTypes.string,
        onChange: PropTypes.func.isRequired,
        maxLength: PropTypes.number,
    };

    componentDidMount() {
        if (this.props.focus) {
            this.input.focus();
        }
    }

    componentWillReceiveProps(nextProps) {
        if (!this.props.focus && nextProps.focus) {
            this.input.focus();
        }
    }

    render() {
        const { value, onChange, disabled, maxLength } = this.props;

        return (
            <input
                ref={(input) => {
                    this.input = input;
                }}
                type="text"
                value={value}
                onChange={(e) => onChange(e.target.value)}
                maxLength={maxLength}
                className={classNames(css.sseed_textline)}
            />
        );
    }
}
