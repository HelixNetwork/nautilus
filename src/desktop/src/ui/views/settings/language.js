import React from 'react';
import css from './settings.scss';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { withI18n } from 'react-i18next';
import { connect } from 'react-redux';
import Button from 'ui/components/button';
import Select from 'ui/components/input/select';
import { I18N_LOCALE_LABELS, I18N_LOCALES } from 'libs/i18n';
import { setLocale } from 'actions/settings';
import i18next from 'libs/i18next';
/**
 * Setting Language component
 */

class SettingsLanguage extends React.PureComponent {
    static propTypes = {
        /** @ignore */
        t: PropTypes.func.isRequired,
        /** @ignore */
        setLocale: PropTypes.func.isRequired,
    };
    state = {
        step: 'language',
        scrollEnd: false,
    };
    changeLocale = (e) => {
        e.preventDefault();

        const { selection } = this.state;

        this.props.setLocale(selection);
        i18next.changeLanguage(selection);

        this.setState({
            selection: null,
        });
    };

    render() {
        const { t, locale } = this.props;
        const { selection } = this.state;
        return (
            <div className={classNames(css.foo_bxx12)}>
                <form className={css.margin_form} onSubmit={(e) => this.changeLocale(e)}>
                    <Select
                        label={t('languageSetup:language')}
                        value={I18N_LOCALE_LABELS[I18N_LOCALES.indexOf(selection || locale)]}
                        onChange={(value) => this.setState({ selection: value })}
                        options={I18N_LOCALES.map((item, index) => {
                            return { value: item, label: I18N_LOCALE_LABELS[index] };
                        })}
                    />
                    <Button style={{ marginLeft: '31vw', marginTop: '4vw' }} type="submit">
                        {t('global:save')}
                    </Button>
                    <div className={classNames(css.spe_bx)}></div>
                </form>
            </div>
        );
    }
}
const mapStateToProps = (state) => ({
    locale: state.settings.locale,
});
const mapDispatchToProps = {
    setLocale,
};
export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(withI18n()(SettingsLanguage));
