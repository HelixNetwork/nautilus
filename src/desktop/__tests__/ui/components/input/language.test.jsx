import React from 'react';
import { shallow } from 'enzyme';

import {LanguageSelect} from 'ui/components/input/language';


const props = {
    setLocale: () => {},
    t: (str) => str
};

let wrapper;
beforeEach(() => {
    wrapper = shallow(<LanguageSelect {...props} />).dive();
});


describe('Test component', () => {
    test('Render the component', () => {
        // expect(wrapper).toMatchSnapshot();
    });
});