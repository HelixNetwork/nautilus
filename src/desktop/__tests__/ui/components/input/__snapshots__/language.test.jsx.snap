// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Render the component 1`] = `undefined`;
